entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -60,27 +60,36 @@
     @Resource(name = AppConfig.BEAN_EM_DATA_SOURCES)
     private DataSources datasources;
 
-    @ApiOperation(value = "Delete an entity", nickname = "deleteEntity", response = java.lang.Void.class)
+    @ApiOperation(value = "Disable an entity", nickname = "disableEntity", response = java.lang.Void.class)
     @RequestMapping(value = "/{type}/{namespace}/{identifier}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<String> deleteEntity(
+    public ResponseEntity<String> disableEntity(
 	    @RequestParam(value = CommonApiConstants.PARAM_WSKEY, required = false) String wskey,
 	    @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
 	    @PathVariable(value = WebEntityConstants.PATH_PARAM_NAMESPACE) String namespace,
 	    @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
-	    HttpServletRequest request) {
-	String entityUri = getEntityUri(type, namespace, identifier.toLowerCase());
-	
-	//TODO create disable entity method in EntityRecordService and implement it accordign to the specs.
-//	long numberDeletedEntities = entityRecordRepository.deleteForGood(entityUri);
-	long numberDeletedEntities = 0;
-	if (numberDeletedEntities == 0) {
-	    return ResponseEntity.notFound()
-		    .header("info:", "There is no entity with the given identifier to be deleted.").build();
-	} else {
-	    String response = "The request is executed successfully. The number of deleted entites is: "
-		    + String.valueOf(numberDeletedEntities);
-	    return ResponseEntity.noContent().header("info:", response).build();
-	}
+	    HttpServletRequest request) throws HttpException {
+    	
+    	String entityUri = getEntityUri(type, namespace, identifier.toLowerCase());
+    	Optional<EntityRecord> existingEntity = entityRecordService.retrieveEntityRecordByUri(entityUri);
+    	if (existingEntity.isPresent()) {
+    		
+			Date timestamp = (existingEntity.get().getEntity().getIsAggregatedBy() != null) ? existingEntity.get().getEntity().getIsAggregatedBy().getModified() : null;
```

Comment:
the first check should be for the disabled flag. I left a comment in the specs which should be clarified today. It is likely that in this case no operation should be performed in the database but the status 410 should be returned..

Reply:
We return the status 410 when the entity is already disabled and the status 404 (not found) if the entity has not been found.